src/io/configreader.hpp

```diff
@@ -748,18 +620,17 @@ void ConfigReader<T>::load(std::string &specifier) {
           model->block_index_map.insert(
               {static_cast<std::string>(heart_outlet_junction_name),
                block_count});
-          block_count++;
-          for (auto heart_outlet_block : closed_loop_config["outlet_blocks"]) {
-            connections.push_back(
-                {heart_outlet_junction_name, heart_outlet_block});
+          for (auto outlet_block :
+               closed_loop_config.get_string_array("outlet_blocks")) {
+            connections.push_back({heart_outlet_junction_name, outlet_block});
+            block_count++;
```

Comment:
This should not be inside this loop because this will increment `block_count` with every connection, instead of incrementing it once for the junction.